pkg/result/processors/autogenerated_exclude.go

```diff
@@ -56,66 +63,96 @@ func (p *AutogeneratedExclude) shouldPassIssue(issue *result.Issue) (bool, error
 		return true, nil
 	}
 
-	if isSpecialAutogeneratedFile(issue.FilePath()) {
+	if !isGoFile(issue.FilePath()) {
 		return false, nil
 	}
 
-	fs, err := p.getOrCreateFileSummary(issue)
-	if err != nil {
-		return false, err
+	// The file is already known.
+	fs := p.fileSummaryCache[issue.FilePath()]
+	if fs != nil {
+		return !fs.generated, nil
 	}
 
+	fs = &fileSummary{}
+	p.fileSummaryCache[issue.FilePath()] = fs
+
+	if issue.FilePath() == "" {
+		return false, errors.New("no file path for issue")
+	}
+
+	if p.strict {
+		var err error
+		fs.generated, err = p.isGeneratedFileStrict(issue.FilePath())
+		if err != nil {
+			return false, fmt.Errorf("failed to get doc of file %s: %w", issue.FilePath(), err)
+		}
+	} else {
+		doc, err := getComments(issue.FilePath())
+		if err != nil {
+			return false, fmt.Errorf("failed to get doc of file %s: %w", issue.FilePath(), err)
+		}
+
+		fs.generated = p.isGeneratedFileLax(doc)
+	}
+
+	p.debugf("file %q is generated: %t", issue.FilePath(), fs.generated)
+
 	// don't report issues for autogenerated files
-	return !fs.isGenerated, nil
+	return !fs.generated, nil
 }
 
-// isGenerated reports whether the source file is generated code.
-// Using a bit laxer rules than https://go.dev/s/generatedcode to
-// match more generated code. See #48 and #72.
-func isGeneratedFileByComment(doc string) bool {
-	const (
-		genCodeGenerated = "code generated"
-		genDoNotEdit     = "do not edit"
-		genAutoFile      = "autogenerated file" // easyjson
-	)
-
+// isGeneratedFileLax reports whether the source file is generated code.
+// Using a bit laxer rules than https://go.dev/s/generatedcode to match more generated code.
```

Comment:
I think we can refer to the `isGeneratedFileStrict` function here:
```suggestion
// The function uses a bit laxer rules than isGeneratedFileStrict to match more generated code.
```